BootloaderCorePkg/Library/PciEnumerationLib/PciResizableBar.c

```diff
@@ -0,0 +1,128 @@
+                        PciIoDevice->Address,
+                        PciIoDevice->ResizableBarOffset,
+                        PciIoDevice->ResizableBarNumber));
+  ResizableBarNumber = MIN (PciIoDevice->ResizableBarNumber, PCI_MAX_BAR);
+  for (Index = 0; Index < ResizableBarNumber; Index++) {
+    Offset = PciIoDevice->ResizableBarOffset + sizeof (PCI_EXPRESS_EXTENDED_CAPABILITIES_HEADER) +
+             sizeof (PCI_EXPRESS_EXTENDED_CAPABILITIES_RESIZABLE_BAR_ENTRY) * Index;
+    Entries[Index].ResizableBarCapability.Uint32 = PciExpressRead32 (PciIoDevice->Address + Offset);
+    Offset += sizeof (PCI_EXPRESS_EXTENDED_CAPABILITIES_RESIZABLE_BAR_CAPABILITY);
+    Entries[Index].ResizableBarControl.Uint32 = PciExpressRead32 (PciIoDevice->Address + Offset);
+  }
+
+  for (Index = 0; Index < ResizableBarNumber; Index++) {
+    //
+    // When the bit of Capabilities Set, indicates that the Function supports
+    // operating with the BAR sized to (2^Bit) MB.
+    // Example:
+    // Bit 0 is set: supports operating with the BAR sized to 1 MB
+    // Bit 1 is set: supports operating with the BAR sized to 2 MB
+    // Bit n is set: supports operating with the BAR sized to (2^n) MB
+    //
+    Capabilities = LShiftU64 (Entries[Index].ResizableBarControl.Bits.BarSizeCapability, 28)
+                   | Entries[Index].ResizableBarCapability.Bits.BarSizeCapability;
+
+    if (ResizableBarOp == PciResizableBarMax) {
+      Bit = HighBitSet64 (Capabilities);
+    } else {
+      ASSERT (ResizableBarOp == PciResizableBarMin);
+      Bit = LowBitSet64 (Capabilities);
+    }
+
+    ASSERT (Bit >= 0);
+
+    Offset = PciIoDevice->ResizableBarOffset + sizeof (PCI_EXPRESS_EXTENDED_CAPABILITIES_HEADER)
+             + Index * sizeof (PCI_EXPRESS_EXTENDED_CAPABILITIES_RESIZABLE_BAR_ENTRY)
+             + OFFSET_OF (PCI_EXPRESS_EXTENDED_CAPABILITIES_RESIZABLE_BAR_ENTRY, ResizableBarControl);
+
+    Entries[Index].ResizableBarControl.Bits.BarSize = (UINT32)Bit;
+    DEBUG ((DEBUG_VERBOSE, "Resizable Bar: Offset = 0x%x, Bar Size Capability = 0x%016lx, New Bar Size = 0x%lx\n",
+                            OFFSET_OF (PCI_TYPE00, Device.Bar[Entries[Index].ResizableBarControl.Bits.BarIndex]),
+                            Capabilities,
+                            LShiftU64 (SIZE_1MB, Bit)));
+    PciExpressWrite32 (PciIoDevice->Address + Offset, Entries[Index].ResizableBarControl.Uint32);
+  }
+  return EFI_SUCCESS;
+}
+
+/**
+  Initialize Resizable BAR
+
+  @param[in,out]  PciIoDevice     Pointer to instance of PCI_IO_DEVICE.
+  @param[in]      Bus             Device Bus NO.
+  @param[in]      Device          Device device NO.
+  @param[in]      Func            Device func NO.
+
+**/
+VOID
+EFIAPI
+InitializeResizeBar (
+  IN OUT  PCI_IO_DEVICE              *PciIoDevice,
+  IN      UINT8                       Bus,
+  IN      UINT8                       Device,
+  IN      UINT8                       Func
```

Comment:
Bus, device and Func are not used in this internal function, better remove them.

Reply:
It is used in the [PciEnumerationLib.c](https://github.com/slimbootloader/slimbootloader/blob/0a9d7749e384370162c70b22368ac1d6661c32db/BootloaderCorePkg/Library/PciEnumerationLib/PciEnumerationLib.c#L446) when enable "SUPPORT_RESIZABLE_BAR" through the board build script. 
Removing the bus,device,func generates conflict with previous declaration of "InitializeResizeBar" from InternalPciEnumerationLib.h

Reply:
This is newly added internal function, not sure why add BDF since they are not used.  If remove BDF, of course remove it from header file and caller at same time in this PR.
BTW, this function is not library API function, it is not necessary to add EFIAPI.

Reply:
ok, removed the BDF from function